tests/tests/da_compression.rs

```diff
@@ -0,0 +1,105 @@
+use core::time::Duration;
+use fuel_core::{
+    combined_database::CombinedDatabase,
+    fuel_core_graphql_api::worker_service::DaCompressionConfig,
+    p2p_test_helpers::*,
+    service::{
+        Config,
+        FuelService,
+    },
+};
+use fuel_core_client::client::{
+    types::TransactionStatus,
+    FuelClient,
+};
+use fuel_core_poa::signer::SignMode;
+use fuel_core_types::{
+    fuel_crypto::SecretKey,
+    fuel_tx::{
+        Input,
+        Transaction,
+    },
+    secrecy::Secret,
+};
+use rand::{
+    rngs::StdRng,
+    SeedableRng,
+};
+
+#[tokio::test]
+async fn can_fetch_da_compressed_block_from_graphql() {
+    let mut rng = StdRng::seed_from_u64(10);
+    let poa_secret = SecretKey::random(&mut rng);
+
+    let db = CombinedDatabase::default();
+    let mut config = Config::local_node();
+    config.consensus_signer = SignMode::Key(Secret::new(poa_secret.into()));
+    config.da_compression = DaCompressionConfig::Enabled(fuel_core_compression::Config {
+        temporal_registry_retention: Duration::from_secs(3600),
```

Comment:
Did you consider testing this mechanism with duration set to `0`?

Reply:
Not yet, but will be done in https://github.com/FuelLabs/fuel-core/issues/2238